test/e2e/storage/testsuites/snapshotsecrets.go

```diff
@@ -0,0 +1,239 @@
+			originalMntTestData = fmt.Sprintf("hello from %s namespace", pvc.GetNamespace())
+			command := fmt.Sprintf("echo '%s' > /mnt/test/data", originalMntTestData)
+
+			RunInPodWithVolume(cs, pvc.Namespace, pvc.Name, "pvc-snapshottable-tester", command, config.ClientNodeSelection)
+			err = e2epv.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, cs, pvc.Namespace, pvc.Name, framework.Poll, framework.ClaimProvisionTimeout)
+			framework.ExpectNoError(err)
+
+			ginkgo.By("checking the claim")
+			// Get new copy of the claim
+			pvc, err = cs.CoreV1().PersistentVolumeClaims(pvc.Namespace).Get(context.TODO(), pvc.Name, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+			// Get the bound PV
+			ginkgo.By("checking the PV")
+			_, err = cs.CoreV1().PersistentVolumes().Get(context.TODO(), pvc.Spec.VolumeName, metav1.GetOptions{})
+			framework.ExpectNoError(err)
+
+		}
+
+		cleanup := func() {
+			// Don't register an AfterEach then a cleanup step because the order
+			// of execution will do the AfterEach first then the cleanup step.
+			// Also AfterEach cleanup registration is not fine grained enough
+			// Adding to the cleanup steps allows you to register cleanup only when it is needed
+			// Ideally we could replace this with https://golang.org/pkg/testing/#T.Cleanup
+
+			// Depending on how far the test executed, cleanup accordingly
+			// Execute in reverse order, similar to defer stack
+			for i := len(cleanupSteps) - 1; i >= 0; i-- {
+				err := tryFunc(cleanupSteps[i])
+				framework.ExpectNoError(err, "while running cleanup steps")
+			}
+
+		}
+		ginkgo.BeforeEach(func() {
+			init()
+		})
+		ginkgo.AfterEach(func() {
+			cleanup()
+		})
+
+		ginkgo.Context("", func() {
+			var (
+				vs     *unstructured.Unstructured
+				secret *v1.Secret
+				sr     *SnapshotResource
+			)
+
+			ginkgo.BeforeEach(func() {
+
+				cleanupSteps = append(cleanupSteps, func() {
+					framework.ExpectNoError(sr.CleanupResource())
+				})
+				// create a secret
+				secret = &v1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Namespace: f.Namespace.Name,
+						Name:      "snapshot-secret",
+					},
+					Data: map[string][]byte{
+						"secret-data": []byte("secret-value-1"),
+					},
+				}
+
+				if secret, err = cs.CoreV1().Secrets(f.Namespace.Name).Create(context.TODO(), secret, metav1.CreateOptions{}); err != nil {
+					framework.Failf("unable to create test secret %s: %v", secret.Name, err)
+				}
+
+				// Create snapshot using secret
+				parameters := map[string]string{
+					CSISnapshotterSecretName:      "snapshot-secret",
+					CSISnapshotterSecretNamespace: f.Namespace.Name,
+				}
+				sr = CreateSnapshotResource(sDriver, config, pattern, pvc.GetName(), pvc.GetNamespace(), parameters)
+				vs = sr.Vs
+			})
+			ginkgo.It("Use secrets field and delete snapshots", func() {
+
+				ginkgo.By("Delete the snapshot")
+				DeleteSnapshotWithSecrets(cs, dc, vs.GetNamespace(), vs.GetName(), framework.Poll, framework.SnapshotDeleteTimeout)
```

Comment:
CleanupResource will take care of snapshot deletion so there's no need to delete snapshot here.  I think this DeleteSnapshotWithSecrets should just be checking if delete snapshot secrets annotation is applied to the VolumeSnapshotContent.